### PR TITLE
Add error boundary

### PR DESCRIPTION
**Pull request by mattupham**
## Purpose? (Required)

Sometimes the screen would go completely white when an error was thrown, add a [React Error Boundary ](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary) to handle uncaught exceptions

## What was modified? (Required)

- Added an error boundary
- Added an error fallback component
- Added bug / support links
- Added i18n

## How was this tested? (Required)

- Manually threw an error within a child component

## Screenshots (Optional)

Screenshot 2023-06-01 at 17 30 58
Screenshot 2023-06-01 at 17 30 50

## Notes for Reviewers (Optional)

- Sentry will be added in a follow up PR

### CODE REVIEW

**Review by daniel-farina — APPROVED** [review]
lgtm, is it possible to test an error message from the previews?

File: packages/web/components/error/error-boundary.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError(_: Error): ErrorBoundaryState {
+    return { hasError: true };
+  }
+
+  componentDidCatch(error: Error, errorInfo: ErrorInfo): void {
+    console.log({ error, errorInfo });
+    // TODO log error to Amplitude in follow up PR
```
Comment (MaxMillington) [inline packages/web/components/error/error-boundary.tsx:27]: and sentry

File: packages/web/components/error/error-boundary.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+interface ErrorBoundaryProps {
+  children: ReactNode;
+  fallback: ReactNode;
+}
+
+interface ErrorBoundaryState {
+  hasError: boolean;
+}
+
+class ErrorBoundary extends React.Component<
```
Comment (MaxMillington) [inline packages/web/components/error/error-boundary.tsx:12]: Can we turn this into a function component with hooks instead?

File: packages/web/components/error/error-boundary.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  ErrorBoundaryProps,
+  ErrorBoundaryState
+> {
+  constructor(props: ErrorBoundaryProps) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError(_: Error): ErrorBoundaryState {
```
Comment (MaxMillington) [inline packages/web/components/error/error-boundary.tsx:21]: side note but it's classic 'react team' MO that they make this huge deal about hooks and then don't build an equivalent for this

**Review by MaxMillington — COMMENTED** [review]
Looks good to me. We just need to add sentry. Getting everything in place for that today

File: packages/web/pages/_app.tsx [hunk 4]
```diff
@@ -147,7 +150,9 @@ function MyApp({ Component, pageProps }: AppProps) {
           transition={Bounce}
         />
         <MainLayout menus={menus}>
-          {Component && <Component {...pageProps} />}
+          <ErrorBoundary fallback={ErrorFallback}>
+            {Component && <Component {...pageProps} />}
+          </ErrorBoundary>
```
[the remark this answers is not in the document]
Comment (MaxMillington) [reply packages/web/pages/_app.tsx:155]: I actually think this is better anyway. Theoretically this wont crash the main layout then (of course we don't have anything wrapping the main layout though)

**Review by MaxMillington — APPROVED** [review]
I think that this is probably ok to merge and then as a followup we can add sentry and change the styling of the error screen. Right now, this will prevent the screen from going white, which is a win

File: packages/web/components/error/error-boundary.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+interface ErrorBoundaryProps {
+  children: ReactNode;
+  fallback: ReactNode;
+}
+
+interface ErrorBoundaryState {
+  hasError: boolean;
+}
+
+class ErrorBoundary extends React.Component<
```
Comment (jonator) [inline packages/web/components/error/error-boundary.tsx:12]: Is there a rationale for using a class component? IMO we should only us FunctionComponent.

File: packages/web/localizations/es.json [hunk 6]
```diff
@@ -461,5 +461,10 @@
   "autonomy": {
     "description": "Explora Órdenes de Límite y Stop presentadas por Autonomy: ",
     "link": "osmosis.autoswap.trade"
-  }
-}
+  },
+  "errors.fallbackText1": "Ocurrió un error",
```
Comment (jonator) [inline packages/web/localizations/es.json:465]: Oh interesting. Seems like inlang doesn't know how to embed a child member to the JSON object

Comment (jonator) [reply packages/web/localizations/es.json:465]: I reported this FYI

**Comment by MaxMillington** [comment]
@mattupham in the future let's make sure to only merge to the 'stage' branch unless it's a hotfix. Especially on a friday

**Pull request merged**